content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,247 @@
+
+You can use the [Kinto Admin](http://kinto.github.io/kinto-admin/) to visually browse and manage your application data.
+
+![Our records in Kinto Admin](images/kinto-admin-webchat-example.png)
+{.image}
+
+With this in mind:
+
+* We will retrieve all messages for a 4815162342 room by requesting `GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records`
+* We will post new messages to the room using `POST https://kinto.discute.re/buckets/chat/collections/4815162342/records`
+
+It's very similar to our first envisioned API calls which were:
+
+* `GET https://api.dicute.re/v1/rooms/4815162342/messages`,
+* `POST https://api.dicute.re/v1/rooms/4815162342/messages`.
+
+## Let’s install Kinto
+
+There are many ways to [install (or deploy) Kinto](https://kinto.readthedocs.io/en/stable/tutorials/install.html):
+
+* [on Heroku](https://kinto.readthedocs.io/en/stable/tutorials/install.html#deploying-on-cloud-providers),
+* [with Docker](https://kinto.readthedocs.io/en/stable/tutorials/install.html#using-docker),
+* [using a Python package](https://kinto.readthedocs.io/en/stable/tutorials/install.html#using-the-python-package).
+
+If you want to play with Kinto, an ephemeral instance is also available on [https://kinto.dev.mozaws.net/v1/](https://kinto.dev.mozaws.net/v1/).
+
+Once deployed, we have access to our `https://kinto.discute.re/v1` default URL that returns:
+
+```json
+{
+    "project_version": "5.0",
+    "settings": {
+        "readonly": false,
+        "batch_max_requests": 25
+    },
+    "capabilities": {
+        "admin": {
+            "description": "Serves the admin console.",
+            "version": "1.5.0",
+            "url": "https://github.com/Kinto/kinto-admin/"
+        },
+        "default_bucket": {
+            "description": "The default bucket is an alias for a personal bucket where collections are created implicitly.",
+            "url": "https://kinto.readthedocs.io/en/latest/api/1.x/buckets.html#personal-bucket-default"
+        }
+    },
+    "project_docs": "https://kinto.readthedocs.io/",
+    "project_name": "kinto",
+    "http_api_version": "1.12",
+    "url": "http://kinto.discute.re/v1/"
+
+}
+```
+
+And that's it! Now we can move to the front-end development phase.
+
+## Viva JavaScript!
+
+In order to get the messages, we will have to write a bit of code to get the room's id through the application URL:
+
+```js
+// https://discute.re/#4815162342  is mapped to the collection 4815162342
+this.collectionName = window.location.hash.slice(1);
+```
+
+Then we need to do a GET request to get the list of all room's messages.
+A simple `GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records` would do the job, but we will rather be using the [kinto-https-js](https://github.com/kinto/kinto-http.js) library that offers a clean API, take a look at this code ma’:
```

Comment:
Of course, I guess we should probably strike it anyhow to be more inclusive, since not everyone was born to a biological mother, grew up with that biological mother, shows off to their mother, etc.